### PR TITLE
[language] more tests for reconfiguration

### PR DESCRIPTION
**Pull request by sblackshear**
Now that functional tests allow us to create validator accounts, we can test more corner cases of the reconfiguration logic.

## Motivation

Make sure reconfiguration code works as expected

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

New tests.

### CODE REVIEW

File: language/functional_tests/tests/testsuite/validator_set/reconfiguration.mvir [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+//! sender: vivian
+import 0x0.LibraSystem;
+import 0x0.ValidatorConfig;
+
+// rotate vivian's pubkey and then run the block prologue. Now, reconfiguration should be triggered.
+main() {
+    let info: LibraSystem.ValidatorInfo;
+
+    ValidatorConfig.rotate_consensus_pubkey(h"40");
+    LibraSystem.block_prologue(2);
```
Comment (davidiw) [inline language/functional_tests/tests/testsuite/validator_set/reconfiguration.mvir]: might be nice to verify that the old key is still in use prior to the block_prologue

File: language/functional_tests/tests/testsuite/validator_set/reconfiguration.mvir [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+// check: ContractEvent {
+// TODO: check for reconfiguration event type once we have it
+// check: EXECUTED
+
+//! new-transaction
+//! sender: vivian
+import 0x0.LibraSystem;
+import 0x0.ValidatorConfig;
+// rotate vivian's pubkey to the same value and run the block prologue. No reconfiguration should be
```
Comment (davidiw) [inline language/functional_tests/tests/testsuite/validator_set/reconfiguration.mvir]: how do we know it didn't happen?

File: language/stdlib/modules/libra_system.mvir [hunk 3]
```diff
@@ -222,8 +238,17 @@ module LibraSystem {
         }
     }
 
-  // Get the address of the i'th validator.
-  public get_ith_validator_address(i: u64): address acquires ValidatorSet {
+    // Get the ValidatorInfo for the ith validator
+    public get_ith_validator_info(i: u64): Self.ValidatorInfo acquires ValidatorSet {
+      let vector_ref: &Vector.T<Self.ValidatorInfo>;
```
Comment (davidiw) [inline language/stdlib/modules/libra_system.mvir]: white space seems inconsistent between 2 and 4 spaces

File: language/functional_tests/tests/testsuite/validator_set/reconfiguration.mvir (hunk 2 above)
Comment (sblackshear) [reply language/functional_tests/tests/testsuite/validator_set/reconfiguration.mvir]: Ah: it's subtle, but the `// check: events []` part after the tx is testing this. I'll add a comment to make this explicit.

File: language/stdlib/modules/libra_system.mvir (hunk 3 above)
Comment (sblackshear) [reply language/stdlib/modules/libra_system.mvir]: Yeah, this is a problem... Hopefully we can add an autoformatter in the forthcoming Move source lang (send pleas to @tnowacki)!

**Comment by sblackshear** [comment]
> Looks nice! Is there a syntax guide on tests in move?

There's not, but we should really add one. This testing framework of @vgao1996's is getting quite powerful. It would be nice to show users how to leverage it. It's so much easier to write tests in this way than it is to manually deploy a bunch of stuff to a local blockchain instance/run test transactions. When I show smart contract devs coming from other platforms, I think they're more impressed with this than with the language...

File: language/functional_tests/tests/testsuite/validator_set/reconfiguration.mvir [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+    info = LibraSystem.get_ith_validator_info(0);
+    assert(*LibraSystem.get_consensus_pubkey(&info) == h"80", 15);
+    // check that the validator set contains Viola's new key
+    info = LibraSystem.get_ith_validator_info(1);
+    assert(*LibraSystem.get_consensus_pubkey(&info) == h"90", 16);
+    return;
+}
+
+// check: ContractEvent {
+// TODO: check for reconfiguration event type once we have it
```
Comment (runtian-zhou) [inline language/functional_tests/tests/testsuite/validator_set/reconfiguration.mvir]: Now that #1261 in landed, you can probably just add the check here. See the new test cases in #1261.

**Comment by vgao1996** [comment]
@sblackshear I heard your message. I'll write something up.
